test/X86/bug-reorder-bb-jrcxz.s

```diff
@@ -0,0 +1,597 @@
+# The test performs a BB reordering with unsupported 
+# instruction jrcxz. Reordering works correctly with the
+# following options: None, Normal or Reverse. Other strategies 
+# are completed with Assertion `isIntN(Size * 8 + 1, Value).
+# The root cause is the distance between BB where one contains 
+# jrcxz instruction.
+# Example: OpenSSL
+# https://github.com/openssl/openssl/blob/master/crypto/bn/asm/x86_64-mont5.pl#L3319
+
+# REQUIRES: system-linux
+
+# RUN: llvm-mc -filetype=obj -triple x86_64-unknown-unknown \
+# RUN:   %s -o %t.o
+# RUN: link_fdata %s %t.o %t.fdata
+# RUN: %host_cc --version
+# RUN: %host_cc %t.o -falign-labels -march=native -o %t.exe -Wl,-q
+
+# RUN:  llvm-bolt %t.exe -o %t.bolted -data %t.fdata \
+# RUN:    -debug -print-disasm -print-all -v=2 -lite=false \
```

Comment:
We don't use -debug on tests because if we build a version of BOLT without debug_info, it will fail the test.